js/classes/entities/midground/enemies/village/snake.js

```diff
@@ -82,13 +68,13 @@ class Snake {
         // HOW SHOULD WE MOVE HIM BASED ON HIS CONDITIONS?
         const mult = (this.facing === "left") ? -1 : 1;
         if (this.action === "slithering") {
-            this.x += mult * Snake.SPEED * GAME.clockTick;
+            this.pos = Vector.add(this.pos, new Vector(mult * Snake.SPEED * GAME.clockTick, 0));
         }
     };
 
     /** Draw the Snake on the canvas. */
     draw() {
-        this.animations[this.facing][this.action].drawFrame(this.x, this.y, Snake.SCALE);
+        this.animations[this.facing][this.action].drawFrame(this.pos, Snake.SCALE);
```

Comment:
The Snake's position should be converted to canvas space before passing to the `drawFrame` method, e.g.: 
`this.animations[this.facing][this.action].drawFrame(Vector.worldToCanvasSpace(this.pos), Snake.SCALE);`